Adds recipe for org-jami-bot

hperrey opened the pull request:
### Brief summary of what the package does

`org-jami-bot` builds upon `jami-bot' and extends it with Org mode capture functionality for text messages and images. It allows to schedule agenda items at specific dates, compose multi-measure captures and capture images -- all by sending a message via the GNU Jami messenger.

### Direct link to the package repository

https://gitlab.com/hperrey/org-jami-bot

### Your association with the package

Author and maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Checklist looks great.  I read through `org-jami-bot-default-setup` and it strikes me that this pattern could potentially (also) be captured by a global minor mode.  Then you wouldn't have to check `if (assoc org-jami-bot-capture-key org-capture-templates)`, since you could just check whether the mode is enabled.  And it would give you an opportunity to also let users turn *off* the behavior when they toggle the mode off (e.g. removing the hooks and so forth).

Let me know what you think, because this isn't a blocker.  Some other small nags:

`org-jami-bot.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- org-jami-bot.el#L251: It's safer to sharp-quote function names; use `#'`
- org-jami-bot.el#L252: It's safer to sharp-quote function names; use `#'`
```

hperrey commented:
Thank you so much for the thorough check and review of the code!

I like the idea of making `org-jami-bot` a global minor mode! However, this would probably imply that `jami-bot` should be a minor mode too -- I put this on the todo list for both packages :)
The check for the capture key would still be a good idea, I believe, as I just recently managed to define two templates for one key leading to unexpected results... but one should probably fail more noticeably.

I fixed the sharp-quoting of the functions -- good catch! I also fixed a bug related to link paths on file captures.

If it is ok for you to go forward with the inclusion in Melpa, then I would make the shift to a minor mode at a later stage. I will have to read up on minor modes first ;)

Thanks and cheers!

riscy commented:
Yes, that's fine and ultimately it's your decision - happy to merge. :)
